AddOns/BasicFinancials/app/src/codeunit/ExperienceTier.Codeunit.al

```diff
@@ -0,0 +1,58 @@
+codeunit 20602 "Experience Tier BF"
+{
+    Access = Internal;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Application Area Mgmt. Facade", 'OnValidateApplicationAreas', '', true, true)]
+    local procedure OnValidateApplicationAreas(ExperienceTierSetup: Record "Experience Tier Setup"; TempApplicationAreaSetup: Record "Application Area Setup")
+    var
+    begin
+        if not ExperienceTierSetup."BF Basic Financials" then
+            exit;
+
+        TempApplicationAreaSetup.TestField("BF Orders", false);
+        TempApplicationAreaSetup.TestField("BF Basic Financials", true);
```

Comment:
This one could an issue. Lets consider the following scenario. The extension is installed, then uninstalled (but not unpublished), then I change Experience Tier. That means in SQL Server I still have ExperienceTierSetup."BF Basic Financials", but the value is False. Now, I try to install the extension again, there is no way to do that.

Reply:
I concur. I have previously informed about the issue. I have asked if there is an uninstalled event trigger, the answer was no.

As I see it, there are two solutions, either the current one where it is necessary to unpublished prior to reinstallation. Otherwise the test must be removed.

What will be the best solution, do you have alternatives?